### PR TITLE
Refs [TASK][37151] Design UI Home Screen

### PR DESCRIPTION
**Pull request by datvt-2504**
## Related Tickets
- [#Task 37151: Design UI Home Screen](https://edu-redmine.sun-asterisk.vn/issues/37151)
## WHAT
- 
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/layout/item_league.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintDimensionRatio="1:1"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/cardViewImage"
+            app:layout_constraintWidth_percent="0.18"
+            tools:src="@tools:sample/avatars" />
+
+<!--        <androidx.constraintlayout.widget.Guideline-->
```
Comment (datdt-1935) [inline app/src/main/res/layout/item_league.xml]: không dùng thì xóa đi nhé

Comment (datvt-2504) [reply app/src/main/res/layout/item_league.xml]: ui t k để í quên mất k xoá :D

File: app/src/main/java/com/suntraining/aesport/ui/binding/ViewBindingAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.suntraining.aesport.ui.binding
+
+import android.view.View
+import androidx.databinding.BindingAdapter
+
+@BindingAdapter("app:visibility")
+fun setVisibility(view: View, visible: Boolean) {
+    view.visibility = if (visible) View.VISIBLE else View.GONE
```
Comment (thonc-2135) [inline app/src/main/java/com/suntraining/aesport/ui/binding/ViewBindingAdapter.kt]:
```suggestion
    view.isVisible = visible
```

File: app/src/main/res/layout/fragment_home.xml [hunk 3]
```diff
@@ -12,7 +13,218 @@
...
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:layout_width="match_parent"
+                android:layout_height="match_parent">
+
+                <androidx.constraintlayout.widget.Guideline
+                    android:id="@+id/guideTop"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:orientation="horizontal"
+                    app:layout_constraintGuide_begin="94dp" />
```
Comment (phucnv-2558) [inline app/src/main/res/layout/fragment_home.xml]: hardcore a

File: app/src/main/res/layout/fragment_home.xml [hunk 4]
```diff
@@ -12,7 +13,218 @@
...
+                    android:layout_marginStart="@dimen/dp_8"
+                    android:layout_marginBottom="@dimen/dp_8"
+                    android:text="@string/text_welcome"
+                    android:textColor="@android:color/white"
+                    android:textSize="@dimen/sp_14"
+                    app:layout_constraintBottom_toTopOf="@+id/viewDivide"
+                    app:layout_constraintStart_toStartOf="@+id/viewWelcome" />
+
+                <androidx.appcompat.widget.AppCompatTextView
+                    android:id="@+id/appCompatTextView2"
```
Comment (phucnv-2558) [inline app/src/main/res/layout/fragment_home.xml]: id rõ ràng a

File: app/src/main/java/com/suntraining/aesport/ui/binding/ImageViewBindingAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.suntraining.aesport.ui.binding
+
+import android.widget.ImageView
+import androidx.databinding.BindingAdapter
+import com.suntraining.aesport.utils.loadImage
+
+@BindingAdapter("app:image")
+fun loadImage(imageView: ImageView, url: String?) =
+    imageView.loadImage(url)
```
Comment (huytq-2422) [inline app/src/main/java/com/suntraining/aesport/ui/binding/ImageViewBindingAdapter.kt]: remove, viết luôn trong Extension 
```suggestion
@BindingAdapter("app:imageUrl")
fun ImageView.loadImage(url: String?)
```

File: app/src/main/java/com/suntraining/aesport/ui/binding/RecyclerViewBinding.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import androidx.databinding.BindingAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.suntraining.aesport.base.UpdateData
+
+@BindingAdapter("app:data")
+fun <T> setDataRecyclerView(recyclerView: RecyclerView, data: List<T>?) {
+    if (recyclerView.adapter is UpdateData<*>) {
+        (recyclerView.adapter as UpdateData<T>).updateData(data)
+    }
+}
```
Comment (huytq-2422) [inline app/src/main/java/com/suntraining/aesport/ui/binding/RecyclerViewBinding.kt]: scope của Project 2 bé, ít hàm data binding nên cũng k cần phải tách ra nhiều file, để 1 file sẽ dễ quản lý hơn 
```suggestion
@BindingAdapter("app:data")
fun <T> RecyclerView.setDataRecyclerView(data: List<T>?) {
    if (recyclerView.adapter is UpdateData<*>) {
        (recyclerView.adapter as UpdateData<T>).updateData(data)
    }
}
```

File: app/src/main/java/com/suntraining/aesport/utils/ImageExtension.kt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.suntraining.aesport.utils
+
+import android.widget.ImageView
+import com.bumptech.glide.Glide
+import com.suntraining.aesport.R
+
+fun ImageView.loadImage(url: String?) {
```
Comment (huytq-2422) [inline app/src/main/java/com/suntraining/aesport/utils/ImageExtension.kt]:
```suggestion

@BindingAdapter(...) 
fun ImageView.loadImage(url: String?) {
```

File: app/src/main/res/layout/item_league.xml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginHorizontal="@dimen/dp_20"
+        android:layout_marginVertical="@dimen/dp_10">
+
+        <com.google.android.material.card.MaterialCardView
+            android:id="@+id/cardViewImage"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            app:cardCornerRadius="@dimen/dp_6"
+            app:cardElevation="@dimen/dp_0"
```
Comment (huytq-2422) [inline app/src/main/res/layout/item_league.xml]:
```suggestion
            app:cardElevation="0dp"
```

**Review by huytq-2422 — APPROVED** [review]

**Pull request merged**